lib/validator.js

```diff
@@ -25,6 +25,26 @@ class Validator {
         return date instanceof Date
       }
     })
+
+    this.ajv.addKeyword({
+      keyword: 'isSet',
+      type: 'object',
+      errors: false,
+      validate: (type, data, schema) => {
+        if (data instanceof Set) {
+          data = Array.from(data)
+        }
+
+        if (schema.isSet) {
+          // undo what we do in convertSchemaToAjvFormat
+          // to prevent infinite recursive call
+          delete schema.isSet
+          schema.type.pop('object')
+        }
+
+        return this.ajv.validate(schema, data)
```

Comment:
It will work if this schema is a Set, but it won't work for any nested schemas.

Reply:
I'm guessing it will also work for nested schemas. It's because the custom keyword validate function is not only called on the top level but every level on the schema if `isSet` is presented.

Added a few unit tests for nested schema in 47a0871f629b103024bcf220005359a108ca6d96 & 8c1d990c9e52e59153495d89933c663f43bda09b!